File: app/src/app/services/api/connection/messages/update.ts
import { Message } from "./message";

export interface Vec3 {
    x: number
    y: number
    z: number
}

export interface GameObject {
    radius?: number
    position?: Vec3
    rotation?: Vec3
    removeable?: boolean
}

export interface Update extends Message {
    tick: number
    time: number
    player: GameObject
    objects: { [key: number]: GameObject }
}

File: server/api/types/connection/update.go
package connection

import (
	"github.com/dayaftereh/discover/server/mathf"
)

type GameObject struct {
	Radius     *float64    `json:"radius"`
	Position   *mathf.Vec3 `json:"position"`
	Rotation   *mathf.Vec3 `json:"rotation"`
	Removeable *bool       `json:"removeable"`
}

type WorldUpdate struct {
	Type    MessageType           `json:"type"`
	Tick    *int64                `json:"tick"`
	Time    *float64              `json:"time"`
	Player  *GameObject           `json:"player"`
	Objects map[int64]*GameObject `json:"objects"`
}